### ISSUE TITLE
The `conda update update` tries to reinstall pip-installed packages

### ISSUE THREAD

**Issue by alexander-smolyakov** [conda/conda#12828]
### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

We have a devcontainer with conda onboard. In a case when conda channels don't contain the latest versions of the Python packages, we have to reinstall them from `pip` to address different security vulnerabilities. 

For example:
We run the following command against the Python distribution that comes with conda:

```Dockerfile
RUN python3 -m pip uninstall --yes \
    requests \
    pyopenssl \
    cryptography \
    && python3 -m pip install --upgrade \
    # https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-32681
    requests \
    # https://github.com/pyca/cryptography/security/advisories/GHSA-5cpq-8wj7-hf2v
    pyopenssl \
    cryptography
```
If we try to run the `conda update conda` command after that, the conda will try to reinstall these packages:

```
vscode ➜ ~ $ conda update conda
Collecting package metadata (current_repodata.json): done
Solving environment: / 
The environment is inconsistent, please check the package plan carefully
The following packages are causing the inconsistency:

  - defaults/linux-64::conda-content-trust==0.1.3=py310h06a4308_0
  - defaults/linux-64::urllib3==1.26.15=py310h06a4308_0
  - defaults/linux-64::conda==23.3.1=py310h06a4308_0
done


==> WARNING: A newer version of conda exists. <==
  current version: 23.3.1
  latest version: 23.5.0

Please update conda by running

    $ conda update -n base -c defaults conda

Or to minimize the number of packages updated during conda update use

     conda install conda=23.5.0



## Package Plan ##

  environment location: /opt/conda

  added / updated specs:
    - conda


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    ca-certificates-2023.05.30 |       h06a4308_0         120 KB
    certifi-2023.5.7           |  py310h06a4308_0         152 KB
    conda-package-handling-2.1.0|  py310h06a4308_0         269 KB
    conda-package-streaming-0.8.0|  py310h06a4308_0          28 KB
    cryptography-39.0.1        |  py310h9ce1e76_2         1.4 MB
    libffi-3.4.4               |       h6a678d5_0         142 KB
    openssl-3.0.9              |       h7f8727e_0         5.2 MB
    pyopenssl-23.0.0           |  py310h06a4308_0          97 KB
    python-3.10.11             |       h955ad1f_3        26.8 MB
    requests-2.29.0            |  py310h06a4308_0          97 KB
    sqlite-3.41.2              |       h5eee18b_0         1.2 MB
    xz-5.4.2                   |       h5eee18b_0         642 KB
    ------------------------------------------------------------
                                           Total:        36.1 MB

The following NEW packages will be INSTALLED:

  cryptography       pkgs/main/linux-64::cryptography-39.0.1-py310h9ce1e76_2 
  pyopenssl          pkgs/main/linux-64::pyopenssl-23.0.0-py310h06a4308_0 
  requests           pkgs/main/linux-64::requests-2.29.0-py310h06a4308_0 

The following packages will be UPDATED:

  ca-certificates                     2023.01.10-h06a4308_0 --> 2023.05.30-h06a4308_0 
  certifi                         2022.12.7-py310h06a4308_0 --> 2023.5.7-py310h06a4308_0 
  conda-package-han~                  2.0.2-py310h06a4308_0 --> 2.1.0-py310h06a4308_0 
  conda-package-str~                  0.7.0-py310h06a4308_0 --> 0.8.0-py310h06a4308_0 
  libffi                                   3.4.2-h6a678d5_6 --> 3.4.4-h6a678d5_0 
  openssl                                 1.1.1t-h7f8727e_0 --> 3.0.9-h7f8727e_0 
  python                                 3.10.10-h7a1cb2a_2 --> 3.10.11-h955ad1f_3 
  sqlite                                  3.41.1-h5eee18b_0 --> 3.41.2-h5eee18b_0 
  xz                                      5.2.10-h5eee18b_1 --> 5.4.2-h5eee18b_0 


Proceed ([y]/n)? 
```

As a result, the `conda update conda` will fail with the well-known error `RemoveError`:

```
Downloading and Extracting Packages
                                                                                                                                                               
Preparing transaction: done                                                                                                                                    
Verifying transaction: failed                                                                                                                                  
                                                                                                                                                               
RemoveError: 'pyopenssl' is a dependency of conda and cannot be removed from                                                                                   
conda's operating environment.                                                                                                                                 
RemoveError: 'requests' is a dependency of conda and cannot be removed from                                                                                    
conda's operating environment. 
```

Is it possible to explicitly set up conda to respect packages installed via `pip` and not try to reinstall them?

### Conda Info

```shell
vscode ➜ ~ $ conda info

     active environment : None
            shell level : 0
       user config file : /home/vscode/.condarc
 populated config files : 
          conda version : 23.3.1
    conda-build version : not installed
         python version : 3.10.10.final.0
       virtual packages : __archspec=1=x86_64
                          __glibc=2.31=0
                          __linux=5.10.102.1=0
                          __unix=0=0
       base environment : /opt/conda  (writable)
      conda av data dir : /opt/conda/etc/conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /opt/conda/pkgs
                          /home/vscode/.conda/pkgs
       envs directories : /opt/conda/envs
                          /home/vscode/.conda/envs
               platform : linux-64
             user-agent : conda/23.3.1 requests/2.31.0 CPython/3.10.10 Linux/5.10.102.1-microsoft-standard-WSL2 debian/11 glibc/2.31
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False
```


### Conda Config

```shell
vscode ➜ ~ $ conda config --show-sources
```


### Conda list

```shell
vscode ➜ ~ $ conda list --show-channel-urls
# packages in environment at /opt/conda:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main    defaults
_openmp_mutex             5.1                       1_gnu    defaults
boltons                   23.0.0          py310h06a4308_0    defaults
brotlipy                  0.7.0           py310h7f8727e_1002    defaults
bzip2                     1.0.8                h7b6447c_0    defaults
ca-certificates           2023.01.10           h06a4308_0    defaults
certifi                   2022.12.7       py310h06a4308_0    defaults
cffi                      1.15.1          py310h5eee18b_3    defaults
charset-normalizer        2.0.4              pyhd3eb1b0_0    defaults
conda                     23.3.1          py310h06a4308_0    defaults
conda-content-trust       0.1.3           py310h06a4308_0    defaults
conda-package-handling    2.0.2           py310h06a4308_0    defaults
conda-package-streaming   0.7.0           py310h06a4308_0    defaults
cryptography              41.0.1                   pypi_0    pypi
idna                      3.4             py310h06a4308_0    defaults
jsonpatch                 1.32               pyhd3eb1b0_0    defaults
jsonpointer               2.1                pyhd3eb1b0_0    defaults
ld_impl_linux-64          2.38                 h1181459_1    defaults
libffi                    3.4.2                h6a678d5_6    defaults
libgcc-ng                 11.2.0               h1234567_1    defaults
libgomp                   11.2.0               h1234567_1    defaults
libstdcxx-ng              11.2.0               h1234567_1    defaults
libuuid                   1.41.5               h5eee18b_0    defaults
ncurses                   6.4                  h6a678d5_0    defaults
openssl                   1.1.1t               h7f8727e_0    defaults
packaging                 23.0            py310h06a4308_0    defaults
pip                       23.0.1          py310h06a4308_0    defaults
pluggy                    1.0.0           py310h06a4308_1    defaults
pycosat                   0.6.4           py310h5eee18b_0    defaults
pycparser                 2.21               pyhd3eb1b0_0    defaults
pyopenssl                 23.2.0                   pypi_0    pypi
pysocks                   1.7.1           py310h06a4308_0    defaults
python                    3.10.10              h7a1cb2a_2    defaults
readline                  8.2                  h5eee18b_0    defaults
requests                  2.31.0                   pypi_0    pypi
ruamel.yaml               0.17.21         py310h5eee18b_0    defaults
ruamel.yaml.clib          0.2.6           py310h5eee18b_1    defaults
setuptools                65.6.3          py310h06a4308_0    defaults
six                       1.16.0             pyhd3eb1b0_1    defaults
sqlite                    3.41.1               h5eee18b_0    defaults
tk                        8.6.12               h1ccaba5_0    defaults
toolz                     0.12.0          py310h06a4308_0    defaults
tqdm                      4.65.0          py310h2f386ee_0    defaults
tzdata                    2023c                h04d1e81_0    defaults
urllib3                   1.26.15         py310h06a4308_0    defaults
wheel                     0.38.4          py310h06a4308_0    defaults
xz                        5.2.10               h5eee18b_1    defaults
zlib                      1.2.13               h5eee18b_0    defaults
zstandard                 0.19.0          py310h5eee18b_0    defaults
```


### Additional Context

_No response_

**Comment by travishathaway** [comment]
Hi @alexander-smolyakov,

Thanks for submitting this issue. To answer you question, no there isn't a way to do this in conda. Furthermore, when you update conda in your base environment, it needs to find the and use the packages that it's compatible with. This is most likely why you are experiencing this downgrade for those packages.

In this case, I would recommend creating a new environment to use in the container (e.g. `conda create --name dev`). This way, when you update conda the packages in that environment will not be affected.

Let me know if this makes sense or was helpful.

**Comment by alexander-smolyakov** [comment]
@travishathaway, thanks for your response! 

The proposed approach to creating a new environment seems not to fit us. This could trigger our security scanners since vulnerable packages will be presented in the container. Thus, I have created two tickets to update vulnerable packages in the conda default channel:

- ContinuumIO/anaconda-issues/issues/13219
- ContinuumIO/anaconda-issues/issues/13220

Could you please let me know if it's possible to speed up the updating process for these packages?

**Comment by travishathaway** [comment]
@alexander-smolyakov,

I've had a chance to speak with those responsible for managing the `defaults` channel. We are currently working on updating the requests package and will have it out as soon as possible (next couple of weeks). As for the cryptography library, the latest version available is `41.0.2` (https://anaconda.org/anaconda/cryptography).

We're investigating the compatibility with pyopenssl.

We appreciate your patience. Please let us know if you have any more questions or concerns.
